pages/docs/guides/resend-webhook-events.mdx

```diff
@@ -0,0 +1,325 @@
+
+```tsx
+import db from "./database";
+
+const invalidateUserEmail = inngest.createFunction(
+ { id: 'invalidate-user-email' },
+ { event: 'resend/email.bounced' },
+ async ({ event }) => {
+   const email = event.data.to[0];
+   const user = await db.users.byEmail(email);
+   if (user) {
+     user.email_status = "invalid";
+     await db.users.update(user);
+   }
+ }
+)
+```
+
+Now that you've seen how to handle Resend events in your application, let's look at a more advanced example.
+
+## Creating a function to send email
+
+Suppose that you want want to send user a welcome email when they sign up to your application.
+
+First, create a helper function called `sendEmail` by adapting the example from the [Resend Next.js Quickstart](https://resend.com/docs/send-with-nextjs):
+
+```tsx
+// resend.ts
+
+import { Resend } from 'resend';
+
+const resend = new Resend(process.env.RESEND_API_KEY);
+
+export async function sendEmail(
+	to: string, 
+	subject: string, 
+	content: React.ReactElement
+) {
+  const { data, error } = await resend.emails.send({
+    from: 'Acme <noreply@acme.dev>',
+    to: [to],
+    subject,
+    react: content
+  });
+
+  if (error) {
+    throw error;
+  }
+
+	return data;
+};
+
+```
+
+Now you're able to send emails! Let's put this new function to use.
+
+The below example assumes that your application receives a `app/signup.completed` event when a new user signs up:
+
+```tsx
+
+import { sendEmail } from "./resend";
+
+const sendWelcomeEmail = inngest.createFunction(
+ { id: 'send-welcome-email' },
+ { event: 'app/signup.completed' },
+ async ({ event }) => {
+	 const { user } = event.data;
+   await sendEmail(user.email, "Welcome to Acme", (
+     <div>
+       <h1>Welcome to ACME, {user.firstName}</h1>
+     </div>
+   ));
+  }
+)
+```
+
+Now that we've mastered the basics of sending email with Resend from an Inngest function, you can build even more complex functionality.
+
+## Sending a delayed follow-up email
+
+Every Inngest function handler comes with an additional [`step` object](https://www.inngest.com/docs/reference/functions/step-sleep-until) which provides tools to create more fine-grained functions. Using `step.run` allows you to encapsulate specific code that will be automatically retried ensuring that issues with one part of your function don't force the entire function to re-run. Additionally, [other tools like `step.sleep`](https://www.inngest.com/docs/reference/functions/step-sleep) are available to extend your app's functionality.
+
+The code below sends a welcome email, then uses `step.sleep` to wait for three days before sending another email offering a free trial:
+
+```tsx
+const sendOnboardingEmails = inngest.createFunction(
+ { id: 'onboarding-emails' },
+ { event: 'app/signup.completed' },
+ async ({ event, step }) => { // ← step is available in the handler's arguments
+   const { user } = event.data
+	 const { email, first_name } = user
+
+   await step.run('welcome-email', async () => {
```

Comment:
Some mixed spaces and tabs here (and in other code snippets) may cause strange formatting issues in the docs. I think this can happen with Notion code snippets. It would be good to go through each of these snippets and make sure each uses the same convention. Mostly elsewhere we use 2 spaces for these JavaScript examples :)